A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,554 @@
+                .data()),
+        kGrpcTraceBinHeaderLen);
+  }
+
+  context::Context Extract(const context::propagation::TextMapCarrier& carrier,
+                           opentelemetry::context::Context& context) {
+    return trace::SetSpan(
+        context, nostd::shared_ptr<Span> sp(new DefaultSpan(
+                     GrpcTraceBinHeaderToSpanContext(absl::Base64Unescape(
+                         carrier.Get("grpc-trace-bin"))))));
+  }
+
+ private:
+  constexpr int kGrpcTraceBinHeaderLen = 29;
+
+  std::array<uint8_t, kGrpcTraceBinHeaderLen> SpanContextToGrpcTraceBinHeader(
+      const opentelemetry::trace::SpanContext& ctx) {
+    std::array<uint8_t, kGrpcTraceBinHeaderLen> header;
+    header[0] = 0;
+    header[1] = 0;
+    ctx.trace_id().CopyBytesTo(&header[2], 16);
+    header[18] = 1;
+    ctx.span_id().CopyBytesTo(&header[19], 8);
+    header[27] = 2;
+    header[28] = ctx.trace_flags().flags();
+    return header;
+  }
+
+  opentelemetry::trace::SpanContext GrpcTraceBinHeaderToSpanContext(
+      nostd::string_view header) {
+    if (header.size() != kGrpcTraceBinHeaderLen || header[0] != 0 ||
+        header[1] != 0 || header[18] != 1 || header[27] != 2) {
+      return SpanContext::GetInvalid();
+    }
+    return SpanContext(TraceId(&header[2], 16), SpanId(&header[19], 8),
+                       TraceFlags(header[28]), /*is_remote*/ true);
+  }
+};
+
+class GrpcTextMapCarrier : public opentelemetry::context::TextMapCarrier {
+ public:
+  GrpcTextMapCarrier(grpc_metadata_batch* metadata) : metadata_(metadata) {}
+
+  nostd::string_view Get(nostd::string_view key) {
+    if (key == "grpc-trace-bin") {
+      return absl::Base64Escape(metadata_->GetStringValue(key).value_or(""));
+    } else if (absl::EndsWith(key, "-bin")) {
+      // Maybe ok to support a custom binary propagator. Needs based64 encoding 
+      // validation if so. Not for now.
+      gpr_log(GPR_ERROR, "Binary propagator other than GrpcTraceBinPropagator is not supported.");
+      return "";
+    }
+    return metadata_->GetStringValue(key);
+  }
+
+  void Set(nostd::string_view key, nostd::string_view value) {
+    if (key == "grpc-trace-bin") {
+      metadata_->Set(
+          grpc_core::GrpcTraceBinMetadata(),
+          grpc_core::Slice::FromCopiedString(absl::Base64Unescape(value)));
+    } else if (absl::EndsWith(key, "-bin")) {
+      gpr_log(GPR_ERROR, "Binary propagator other than GrpcTraceBinPropagator is not supported.");
+      return;
+    } else {
+      // A propagator other than GrpcTraceBinTextMapPropagator was used.
+      metadata_->Append(key, grpc_core::Slice::FromCopiedString(value));
+    }
+  }
+
+ private:
+  grpc_metadata_batch* metadata_;
+};
+
+}  // namespace internal
+
+std::unique_ptr<opentelemetry::context::TextMapCarrier>
+MakeGrpcTraceBinTextMapPropagator() {
+  return std::make_unique<internal::GrpcTraceBinTextMapPropagator>();
+}
+
+}  // namespace grpc
+
+```
+
+### gRPC OpenTelemetry Tracing API
```

Comment:
I think this subsection should be the first one in the "proposal" section, because it describes the main new API that we're adding here.